tests/e2e/odh_manager_test.go

```diff
@@ -76,3 +99,216 @@ func (tc *OperatorTestCtx) ValidateOwnedCRDs(t *testing.T) {
 		})
 	}
 }
+
+// Scale the operator for multiple replicas, validate pod leader election and behavior.
+func (tc *OperatorTestCtx) ValidateMultiReplicaLeaderElectionAndRecovery(t *testing.T) {
+	t.Helper()
+
+	// Create DSCI
+	tc.EventuallyResourceCreatedOrUpdated(
+		WithObjectToCreate(CreateDSCI(tc.DSCInitializationNamespacedName.Name, tc.AppsNamespace)),
+		WithCondition(jq.Match(`.status.phase == "%s"`, status.ConditionTypeReady)),
+		WithCustomErrorMsg("Failed to create DSCInitialization resource %s", tc.DSCInitializationNamespacedName.Name),
+	)
+
+	// Fetching the Platform release name
+	platform := tc.FetchPlatformRelease()
+	var selector labels.Selector
+
+	// Determine selector based on platform.
+	switch platform {
+	case cluster.SelfManagedRhoai, cluster.ManagedRhoai:
+		selector = labels.SelectorFromSet(labels.Set{
+			"name": "rhods-operator",
+		})
+	case cluster.OpenDataHub:
+		selector = labels.SelectorFromSet(labels.Set{
+			"control-plane": "controller-manager",
+		})
+	}
+
+	// Find and delete the current leader pod
+	leaderPodName := tc.findLeaderPodFromLeases()
+	require.NotEmpty(t, leaderPodName, "Failed to find leader pod from lease resources")
```

Comment:
```suggestion
        tc.g.Expect(leaderPodName).ShouldNot(BeEmpty(), "Failed to find leader pod from lease resources")
```
Let's use gomega. I'm not sure how this library got introduced. 

Please also replace the other occurrence of `require` and remove the import.
```
require.NoError(t, err, "Failed to initialize test context")
```
with 
```
g.Expect(err).ShouldNot(HaveOccurred(), "Failed to initialize test context")
```